assimilation_code/modules/assimilation/filter_mod.f90

```diff
@@ -394,29 +400,31 @@ subroutine filter_main()
 
 call trace_message('Before initializing inflation')
 
-call validate_inflate_options(inf_flavor, inf_damping, inf_initial_from_restart, &
+call parse_inflation_string(inf_flavor,inflation_flavor)
+
+call validate_inflate_options(inflation_flavor, inf_damping, inf_initial_from_restart, &
    inf_sd_initial_from_restart, inf_deterministic, inf_sd_max_change,            &
    do_prior_inflate, do_posterior_inflate, output_inflation, compute_posterior)
 
 ! Initialize the adaptive inflation module
-call adaptive_inflate_init(prior_inflate, inf_flavor(1), inf_initial_from_restart(1), &
+call adaptive_inflate_init(prior_inflate, inflation_flavor(1), inf_initial_from_restart(1), &
    inf_sd_initial_from_restart(1), output_inflation, inf_deterministic(1),            &
    inf_initial(1), inf_sd_initial(1), inf_lower_bound(1), inf_upper_bound(1),         &
    inf_sd_lower_bound(1), inf_sd_max_change(1), state_ens_handle,                     &
    allow_missing, 'Prior')
 
-call adaptive_inflate_init(post_inflate, inf_flavor(2), inf_initial_from_restart(2),  &
+call adaptive_inflate_init(post_inflate, inflation_flavor(2), inf_initial_from_restart(2),  &
    inf_sd_initial_from_restart(2), output_inflation, inf_deterministic(2),            &
    inf_initial(2),  inf_sd_initial(2), inf_lower_bound(2), inf_upper_bound(2),        &
    inf_sd_lower_bound(2), inf_sd_max_change(2), state_ens_handle,                     &
    allow_missing, 'Posterior')
 
 if (do_output()) then
-   if (inf_flavor(1) > 0 .and. inf_damping(1) < 1.0_r8) then
+   if (inflation_flavor(1) > NO_INFLATION .and. inf_damping(1) < 1.0_r8) then
```

Comment:
Why not also set 1 to PRIOR and 2 to POSTERIOR here?

Reply:
That's just a test to see if you were actually looking  :)    ... clearly missed the one in adaptive_inflate_mod.f90 and didn't know how far to go in the filter_mod.f90 ... there are MANY places  where PRIOR and POSTERIOR could be used.
I had a bit of a dilemma using PRIOR and POSTERIOR for the loop bounds, that only works if PRIOR and POSTERIOR are adjacent numbers ... and actually ONLY if PRIOR and POSTERIOR and 1 and 2, respectively.  If, for some reason, someone wanted to change PRIOR, POSTERIOR to 3 and 7 (or something) the whole system fails.  I've put in a comment to mention that the values of PRIOR and POSTERIOR MUST be 1 and 2 - which clears the way for using PRIOR and POSTERIOR for loop bounds.

Reply:
Done.